fenix/app/src/main/java/org/mozilla/fenix/shopping/store/ReviewQualityCheckState.kt

```diff
@@ -72,13 +69,19 @@ sealed interface ReviewQualityCheckState : State {
              */
             data class AnalysisPresent(
                 val productId: String,
-                val reviewGrade: Grade,
+                val reviewGrade: Grade?,
                 val needsAnalysis: Boolean,
-                val adjustedRating: Float,
+                val adjustedRating: Float?,
                 val productUrl: String,
-                val highlights: Map<HighlightType, List<String>>?,
+                val highlights: SortedMap<HighlightType, List<String>>?,
                 val recommendedProductState: RecommendedProductState = RecommendedProductState.Initial,
-            ) : ProductReviewState
+            ) : ProductReviewState {
+                init {
+                    require(!(highlights == null && reviewGrade == null && adjustedRating == null)) {
+                        "AnalysisPresent state should not be created with these 3 null values"
```

Comment:
potential wording improvement:

```suggestion
                        "AnalysisPresent state should only be created when at least one of these parameters is not null"
```

It might also help the `require` if you include which of the three parameters were null in the message.

Reply:
thanks, updated!